identity/src/keypair.rs

```diff
@@ -343,21 +343,36 @@ impl Keypair {
         }
     }
 
-    /// Deterministically derive a new secret from this [`Keypair`], taking into account the provided domain.
-    ///
-    /// This works for all key types except RSA where it returns `None`.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// # fn main() {
-    /// # use libp2p_identity as identity;
-    ///
-    /// let key = identity::Keypair::generate_ed25519();
-    ///
-    /// let new_key = key.derive_secret(b"my encryption key").expect("can derive secret for ed25519");
-    /// # }
-    /// ```
+    #[doc = r##"
+    Deterministically derive a new secret from this [`Keypair`], taking into account the provided domain.
+
+    This works for all key types except RSA where it returns `None`.
+
+    # Example
+
+    "##]
+    #[cfg_attr(
+        feature = "rand",
+        doc = r##"
+    ```
+    "##
+    )]
+    #[cfg_attr(
+        not(feature = "rand"),
+        doc = r##"
+    ```ignore
+    "##
+    )]
+    #[doc = r##"
+    # fn main() {
+    # use libp2p_identity as identity;
+    let key = identity::Keypair::generate_ed25519();
+
+    let new_key = key.derive_secret(b"my encryption key").expect("can derive secret for ed25519");
+    # }
+    ```
+    "##
+    ]
```

Comment:
`cargo test` runs it as a doctest, and fails when rand feature is not enabled. It can be set to `ignore` statically to make it simpler, but it would mean that this function is not tested anymore (cannot check now if it is tested elsewhere).

Reply:
We always test with all features so it should still be tested, right?

Reply:
Well, it is not very convenient if one works with particular crate and wants to test it with particular flag combination.  I've reverted the comment to the original version, however, I hope it will work for you.